x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -368,15 +363,15 @@ export class CsvGenerator {
       }
     } catch (err) {
       this.logger.error(err);
-      if (err instanceof KbnServerError && err.errBody) {
-        throw JSON.stringify(err.errBody.error);
-      }
-
-      if (err instanceof esErrors.ResponseError && [401, 403].includes(err.statusCode ?? 0)) {
-        reportingError = new AuthenticationExpiredError();
-        warnings.push(i18nTexts.authenticationError.partialResultsMessage);
-      } else {
-        throw new UnknownError(err.message);
+      if (err instanceof esErrors.ResponseError) {
+        if ([401, 403].includes(err.statusCode ?? 0)) {
+          reportingError = new AuthenticationExpiredError();
+          warnings.push(i18nTexts.authenticationError.partialResultsMessage);
+        } else {
+          warnings.push(i18nTexts.esErrorMessage(err.statusCode ?? 0, String(err.body)));
+        }
+      } else if (!this.maxSizeReached) {
```

Comment:
Is this intended to suppress all the errors when we've got enough data? Maybe we should keep warnings to avoid confusion during debugging?

Reply:
I think event of the max size being reached isn't relevant to this part of the code. It could just be:
```suggestion
      } else {
```

Reply:
Ah good catch, this was when I mistakenly thought `tryAppend` would throw for maxSizeReached